train.py

```diff
@@ -8,44 +8,61 @@
 from pytorch_lightning.callbacks import ModelCheckpoint, EarlyStopping, LearningRateLogger
 from pytorch_lightning.loggers import WandbLogger
 
-from configs import get_code2seq_default_config, get_code2seq_test_config
+from configs import (
+    get_code2seq_default_config,
+    get_code2seq_test_config,
+    get_code2class_test_config,
+    get_code2class_default_config,
+)
 from dataset import Vocabulary
-from model import Code2Seq
+from model import Code2Seq, Code2Class
 
 DATA_FOLDER = "data"
 SEED = 7
 
 
-def train(dataset_name: str, num_workers: int = 0, is_test: bool = False, resume_from_checkpoint: str = None):
+def train(
+    dataset_name: str, model_name: str, num_workers: int = 0, is_test: bool = False, resume_from_checkpoint: str = None
+):
     seed_everything(SEED)
     dataset_main_folder = join(DATA_FOLDER, dataset_name)
     vocab = Vocabulary.load(join(dataset_main_folder, "vocabulary.pkl"))
 
-    config_function = get_code2seq_test_config if is_test else get_code2seq_default_config
-    config = config_function(dataset_main_folder)
-
-    model = Code2Seq(config, vocab, num_workers)
+    if model_name == "code2seq":
+        config_function = get_code2seq_test_config if is_test else get_code2seq_default_config
+        hyperparams, encoder_config, decoder_config = config_function(dataset_main_folder)
+        model = Code2Seq(hyperparams, vocab, num_workers, encoder_config=encoder_config, decoder_config=decoder_config)
+    elif model_name == "code2class":
+        config_function = get_code2class_test_config if is_test else get_code2class_default_config
+        hyperparams, encoder_config, decoder_config = config_function(dataset_main_folder)
+        model = Code2Class(
+            hyperparams, vocab, num_workers, encoder_config=encoder_config, decoder_config=decoder_config
+        )
+    else:
+        raise ValueError(f"Model {model_name} is not supported")
 
     # define logger
     wandb_logger = WandbLogger(project=f"code2seq-{dataset_name}", log_model=True, offline=is_test)
     wandb_logger.watch(model)
-    wandb_logger.log_hyperparams(asdict(config))
+    wandb_logger.log_hyperparams(asdict(hyperparams))
+    wandb_logger.log_hyperparams(asdict(encoder_config))
+    wandb_logger.log_hyperparams(asdict(decoder_config))
```

Comment:
You can remove this because we have `self.save_hyperparameters()` in each model. At the moment, parameters are logged twice